Technomancy/tower.ts

```diff
@@ -197,49 +315,142 @@
+            if (!adventure_data["tower_ascension"]) {
+                return "10 more autobuild spaces";
+            }
+            else {
+                return "nothing";
+            }
+        },
         reward: function () { }
     }, 
     { /* Boss 26 */
         "boss": "A Very Large Telescope",
         "text": "It's used for finding exoplanets. Also, you're fighting it now. ",
-        "reward_text": "an extra autobuild slot for every tower boss you kill",
+        get reward_text() {
+            if (!adventure_data["tower_ascension"]) {
+                return "an extra autobuild slot for every tower boss you kill";
+            } else {
+                return "an extra autobuild slot for every new tower boss you  kill";
+            }
+        },
         reward: function () { }
     }, 
     { /* Boss 27 */
         "boss": "A globe",
         "text": "What's so tough about just a regular globe? Well, maybe the fact that this one is to scale. 1:1 scale that is. ",
-        "reward_text": "a lot more mana on every prestige",
+        get reward_text() {
+            if (!adventure_data["tower_ascension"]) {
+                return "a lot more mana on every prestige";
+            }
+            else {
+                return "a lot more mana on every prestige, again";
+            }
+        },
         reward: function () { }
     }, 
     { /* Boss 28 */
         "boss": "a little swimmy fishy",
         "text": "It's just swimming around. ",
-        "reward_text": "being able to kill the environment faster",
+        get reward_text() {
+            if (!adventure_data["tower_ascension"]) {
+                return "being able to kill the environment faster";
+            }
+            else {
+                return "being able to kill the environment faster, again";
+            }
+        },
         reward: function () { }
     }, 
     { /* Boss 29 */
         "boss": "a little golden swimmy fishy",
         "text": "It's just swimming around. It's also very shiny. ",
-        "reward_text": "being able to kill the economy faster",
+        get reward_text() {
+            if (!adventure_data["tower_ascension"]) {
+                return "being able to kill the economy faster";
+            }
+            else {
+                return "being able to kill the economy faster, again";
+            }
+        },
         reward: function () { }
     }, 
     { /* Boss 30 */
         "boss": "a little sparkly swimmy fishy",
         "text": "It's just swimming around. It's also very extremely shiny. ",
-        "reward_text": "even more mana per prestige",
+        get reward_text() {
+            if (!adventure_data["tower_ascension"]) {
+                return "even more mana per prestige";
+            }
+            else {
+                return "even more mana per prestige, again";
+            }
+        },
+        reward: function () { }
+    },
+    { /* Boss Repeat, for extra levels */
+        get boss() {
```

Comment:
I'm not sure if this is quite how I want to do ascension - I'm thinking continue the tower like normal (i.e. hardcoded boss text/rewards), and cap ascensions at however many bosses I have coded. 

Maybe in the future when people have insane amounts of essence auto-generation will be needed, but not yet.

Reply:
Actually.  The intention here is that you can just add more hardcoded bosses before the final two elements and then the penultimate element will fill in the gaps.  The final boss is movable and will shift to later elements.
I'm going to update things so that this is more explicit and the height of the first tower is separate from the length of the TOWER_DATA array.

Reply:
Only issue is that adding hardcoded bosses before the final 2 will add them to pre-ascension tower. So yeah, gotta separate it from TOWER_DATA.length to fix that.